qiskit/providers/__init__.py

```diff
@@ -184,37 +184,60 @@ def backends(self, name=None, **kwargs):
+:attr:`~qiskit.providers.BackendV2.target` property to define the model of the
+backend for the compiler, a :attr:`~qiskit.providers.BackendV2.max_circuits`
+property to define a limit on the number of circuits the backend can execute in
+a single batch job (if there is no limit ``None`` can be used), a
+:meth:`~qiskit.providers.BackendV2.run` method to accept job submissions, and
+a :obj:`~qiskit.providers.BackendV2._default_options` method to define the
+user configurable options and their default values. For example, a minimum working
+example would be something like::
+
+    from qiskit.providers import BackendV2 as Backend
+    from qiskit.transpiler import Target
     from qiskit.providers import Options
+    from qiskit.circuit import Parameter, Measure
+    from qiskit.circuit.library import PhaseGate, SXGate, UGate, CXGate, IGate
 
 
     class Mybackend(Backend):
 
-        configuration = {
-            'backend_name': 'my_backend',
-            'backend_version': '1.0.0',
-            'n_qubits': 5,
-            'basis_gates': ['p', 'sx', 'u', 'cx', 'id'],
-            'coupling_map': [[0, 1], [1, 2], [2, 3], [3, 4]],
-            'simulator': False,
-            'local': False,
-            'conditional': True,
-            'open_pulse': False,
-            'memory': True,
-            'max_shots': 8196,
-            'gates': []
-        }
+        def __init(self):
+            super().__init__()
 
-        def __init(self, provider):
-            super().__init__(
-                configuration=BackendConfiguration.from_dict(self.configuration),
-                provider=provider)
+            # Create Target
+            self._target = Target("Target for My Backend")
+            # Instead of None for this and below instructions you can define
+            # a qiskit.transpiler.InstructionProperties object to define properties
+            # for an instruction.
+            lam = Parameter("λ")
+            p_props = {(qubit,): None for qubit in range(5)}
+            self._target.add_instruction(PhaseGate(lam), p_props)
+            sx_props = {(qubit,): None for qubit in range(5)}
+            self._target.add_instruction(SXGate(), sx_props)
+            phi = Parameter("φ")
+            theta = Parameter("ϴ")
+            u_props = {(qubit,): None for qubit in range(5)}
+            self._target.add_instruction(UGate(theta, phi, lam), u_props)
+            cx_props = {edge: None for edge in [(0, 1), (1, 2), (2, 3), (3, 4)]}
+            self._target.add_instruction(CXGate(), cx_props)
+            meas_props = {(qubit,): None for qubit in range(5)}
+            self._target.add_instruction(Measure(), meas_props)
+            id_props = {(qubit,): None for qubit in range(5)}
+            self._target.add_instruction(IGate(), id_props)
+
+            # Set option validators
+            self.options.set_validator("shots", (1, 4096))
+            self.options.set_validator("measure", bool)
```

Comment:
```python
KeyError: "Field 'measure' is not present in this options object"
```

Reply:
Oops, yeah it should be `"memory"` not `"measure"`. Fixed in the current revision (just forced pushed as part of a rebase too)